### DISCUSSION TITLE
Nested Components and Shared Properties

### DISCUSSION THREAD

**Discussion by gofish543** [livewire/livewire#2877] [1. Help]
I have a problem regarding Livewire and nested components.

I have a form wizard called `Supplier` which has two child components `Meta` and `States`.
What is important to know is the `States` component requires information from the `Meta` component after the data has been entered.

Is there a way in Livewire to have a parent wizard handle steps of a form and allow them to pass data between eachother?

Code
```
<?php

namespace App\Http\Livewire;

use App\Http\Livewire\Supplier\Meta;
use App\Http\Livewire\Supplier\States;
use Livewire\Component;

class SupplierWizard extends Component
{
    public Meta $meta;

    public States $states;

    public function mount(Meta $meta, States $states)
    {
        $this->meta = $meta;
        $this->states = $states;
    }

    public function render()
    {
        return view('supplier.wizard');
    }
}
```
With the view `supplier.wizard`
```
@extends('layouts.public')

@php $title = 'fdsa'; @endphp

@section('scenario::title')
    Create {{ $title }}
@endsection

@section('content')
    <div class="flex flex-col mb-12">
        <form>
            <livewire:$meta/> ... OR?
            <livewire:component=$meta/> ... OR?
            <livewire:supplier.meta/> ... Not this because it creates a new instance outside of the parent.
        </form>
    </div>
@endsection
```

**Comment by gwleuverink** [comment]
Props passed to components are not reactive when the parent updates it like they are in Vue. The easiest way to do it is probably to restructure the code into traits that share the same component.

There is also a third party state library that could do it called sprucewire. Cool as that is it most likely it is not needed. Better keep things simple.
